### PR TITLE
Protocol action "of" validation

### LINKED ISSUE #507: Missing `of` keyword in protocol action rules leads to confusing error
When a protocol definition has a `recipient` or `author` rule, the `of` keyword is required. But the error that the JSON schema throws is difficult for people to understand. `/descriptor/definition/structure/album/$actions/0/who: must be equal to one of the allowed values`

We should either alter the JSON schema to throw a friendlier message, or add validation in plain javascript.

JSON Schema: https://github.com/TBD54566975/dwn-sdk-js/blob/main/json-schemas/interface-methods/protocol-rule-set.json#L24-L74

Protocol Definition javascript validation: https://github.com/TBD54566975/dwn-sdk-js/blob/main/src/interfaces/protocols-configure.ts#L44

```javascript
// correct
[{
  who: 'anyone',
  can: 'read'
},
{
  who: 'recipient',
  of: 'foo',
  can: 'read'
},
{
  who: 'author',
  of: 'foo',
  can: 'read'
}]

// incorrect
[{
  who: 'recipient',
  can: 'read'
},
{
  who: 'author',
  can: 'read'
},
{
  who: 'anyone',
  of: 'foo',
  can: 'read'
}]
```

### PR DESCRIPTION
**Pull request by flothjl**
Fixes #507 
Moving protocol action rule "of" validation to javascript to improve error messaging. 

Debated trying to leverage an ajv plugin to help improve error messaging during schema validation, but a couple of things stood out:

1. Would require changing configurations for `Ajv` instance to return all errors in a validation instead of stopping after the first error. 
2. A lot of the groundwork for this type of validation on `$actions` was added recently with #512

### CODE REVIEW

**Comment by EbonyLouis** [comment]
@flothjl  Just a heads-up: To be eligible for the official Hacktoberfest prizes, PRs need to be merged on or after **October 1st.** Would you like to wait until then or merge it now?  

> to be eligible for Hacktoberfest's official prizes make sure you're registered, registration [is now open.](https://hacktoberfest.com/auth/) Reach out if you need any help 🥳

**Review by diehuxx — DISMISSED** [review]
Thanks @flothjl -- @thehenrytsai and I made a small change and merged main. Like @EbonyLouis said, if you want to be part of the Hacktoberfest thing, we'll wait to merge this until Sunday Oct 1st. Or feel free to merge it sooner, it's up to you.

**Comment by flothjl** [comment]
@diehuxx @EbonyLouis I'm really okay either way here. It doesn't look like I have rights to merge. If waiting until 10/1 helps TBD's Hacktoberfest 🍁 standings I can just reopen the PR on Sunday, otherwise I'm fine if y'all just merge now :). Just let me know.

**Review by diehuxx — APPROVED** [review]

**Pull request merged**
